x-pack/plugins/canvas/public/components/link/link.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, MouseEvent } from 'react';
+import PropTypes from 'prop-types';
+import { EuiLink, EuiLinkProps } from '@elastic/eui';
+
+import { ComponentStrings } from '../../../i18n';
+
+const { Link: strings } = ComponentStrings;
+
+const isModifiedEvent = (ev: MouseEvent) =>
+  !!(ev.metaKey || ev.altKey || ev.ctrlKey || ev.shiftKey);
+
+interface Props {
+  name: string;
+  params: Record<string, any>;
+}
+
+export const Link: FC<Props & EuiLinkProps> = (
+  { onClick, target, name, params, children, ...linkArgs },
+  { router }
```

Comment:
I added a RouterContextProvider and started stubbing out a type for the router as well. You could switch to using `useContext` or at least use the type and add any aditional properties used here to it.

Reply:
@crob611 Could you perhaps push a change for my branch?  Not quite following...